pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -38,22 +37,47 @@ const (
 	remediationStrategyAnnotation = "healthchecking.openshift.io/strategy"
 	remediationStrategyReboot     = mhcv1beta1.RemediationStrategyType("reboot")
 	timeoutForMachineToHaveNode   = 10 * time.Minute
+	machineNodeNameIndex          = "machineNodeNameIndex"
 )
 
 // Add creates a new MachineHealthCheck Controller and adds it to the Manager. The Manager will set fields on the Controller
 // and start it when the Manager is started.
 func Add(mgr manager.Manager, opts manager.Options) error {
-	r := newReconciler(mgr, opts)
+	r, err := newReconciler(mgr, opts)
+	if err != nil {
+		return fmt.Errorf("error building reconciler: %v", err)
+	}
 	return add(mgr, r, r.mhcRequestsFromMachine, r.mhcRequestsFromNode)
 }
 
 // newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager, opts manager.Options) *ReconcileMachineHealthCheck {
+func newReconciler(mgr manager.Manager, opts manager.Options) (*ReconcileMachineHealthCheck, error) {
+	if err := mgr.GetCache().IndexField(&mapiv1.Machine{},
```

Comment:
Don't matter really, so not a blocker, but I feel like this might belong in `Add()` or `add()` instead of here.